docs/UPGRADING.md

```diff
@@ -0,0 +1,254 @@
+
+Areas where the changes are needed:
+
+- You stop using `TaskExecutor` and switch to `GolemNetwork` instead.
+- Be explicit about the expected computation time and pricing strategy so that `golem-js` can estimate the budget.
+- You reach for an exe-unit representation with the `ResourceRental.getExeUnit` method and call your commands via the provided `ExeUnit` instance.
+
+**Before:**
+
+```ts
+// before
+import { TaskExecutor } from "@golem-sdk/golem-js";
+
+(async function main() {
+  const executor = await TaskExecutor.create("golem/alpine:latest");
+  try {
+    await executor.run(async (ctx) => console.log((await ctx.run("echo 'Hello World'")).stdout));
+  } catch (error) {
+    console.error("Failed to execute work:", error);
+  } finally {
+    await executor.shutdown();
+  }
+})();
+```
+
+**After:**
+
+```ts
+// after
+import { GolemNetwork } from "@golem-sdk/golem-js";
+
+(async function main() {
+  const glm = new GolemNetwork();
+  try {
+    await glm.connect();
+
+    const retnal = await glm.oneOf({
+      order: {
+        demand: {
+          workload: { imageTag: "golem/alpine:latest" },
+        },
+        // You have to be now explicit about about your terms and expectatios from the market
+        market: {
+          rentHours: 5 / 60,
+          pricing: {
+            model: "linear",
+            maxStartPrice: 0.5,
+            maxCpuPerHourPrice: 1.0,
+            maxEnvPerHourPrice: 0.5,
+          },
+        },
+      },
+    });
+
+    // You will work with exe-unit objects instead of "executor"
+    await rental
+      .getExeUnit()
+      .then((exe) => exe.run("echo 'Hello World'"))
+      .then((res) => console.log(res.stdout));
+  } catch (error) {
+    console.error("Failed to execute work:", error);
+  } finally {
+    await glm.disconnect();
+  }
+})();
+```
+
+#### Repeating commands with `map` of different parameter values
+
+Areas where the changes are needed:
+
+- Instead of referring the `TaskExecutor` instance in the `map` function, you need to switch to the `ExeUnit` instance provided by the `ResourceRental`
+
+**Before:**
+
+```ts
+// before
+import { GolemNetwork } from "@golem-sdk/golem-js";
+
+(async function main() {
+  const executor = await GolemNetwork.create("golem/alpine:latest");
+  try {
+    const inputs = [1, 2, 3, 4, 5];
+
+    const results = await Promise.allSettled(
+      inputs.map((input) => executor.run((ctx) => ctx.run(`echo 'Hello ${input}`))),
+    );
+
+    const responses = results.map((p) => (p.status === "fulfilled" ? p.value.stdout : null)).filter((v) => v !== null);
+
+    console.log(responses);
+  } catch (error) {
+    console.error("Failed to execute work:", error);
+  } finally {
+    await executor.shutdown();
+  }
+})();
+```
+
+```ts
+// after
+import { GolemNetwork } from "@golem-sdk/golem-js";
+
+(async function main() {
+  const glm = new GolemNetwork();
+  try {
+    await glm.connect();
+
+    const retnal = await glm.oneOf({
```

Comment:
TE by default does this work in parallel, but the provided snipped does it sequentially on one machine. We should promote `manyOf` for cases where multiple things need to be done at the same time, like you do later on

Reply:
You're right, TE with `maxParallelTasks: 5 (the default)` will execute the tasks concurrently, while the new snippet will execute them in a sequential manner.

Nevertheless, the intent of this section was to show 1:1 transition between snippets specifically to show how the `Array.map` can be still used to execute a series of similar commands with different arguments. I'd like to keep the focus on this aspect.

Your observation is valueable and I will honour that by placing a note in the upgrade gude that compared to TE this is going to be executed sequentially, and that for parallel execution people should refer to `manyOf` example.

On a different note, it might be worth considering to fix the naming of the params for `manyOf` -> instead of `concurrency` I would simply use `count`.

Why? Concurrency and parallelism are two different terms. Technically we run parallel distributed "computations", so actually `parallelism` is the word we should use instead of `concurrency` (running on the same thread). 

I will create a ticket for rename of `oder.concurrency` to `order.count` for `manyOf`.